### ISSUE TITLE
`Required environment variable '_APPSIGNAL_PUSH_API_KEY' not present` on OTP 21

### ISSUE THREAD

**Issue by jeffkreeftmeijer** [appsignal/appsignal-elixir#365]
We’re investigating an issue on OTP 21 where the agent doesn’t properly receive the app’s configuration, resulting in log output like this;

```
WARNING: Error when reading appsignal config, appsignal not starting: Required environment variable '_APPSIGNAL_PUSH_API_KEY' not present
```

We’re also seeing this on [CI](https://travis-ci.org/appsignal/appsignal-elixir/jobs/396025746).

**Comment by jeffkreeftmeijer** [comment]
The issue is caused by [a change in OTP 21](https://github.com/erlang/otp/pull/1672), which prevents us from setting environment variables the agent requires to function. We’re working on exposing an API in the agent that allows us to configure it directly without depending on environment variables, which we're hoping to release soon.

We'll keep this issue updated.

**Comment by jeffkreeftmeijer** [comment]
We've exposed the configuration API in the agent, and we're working on a new version of the Elixir integration to use that and remove the dependency on environment variables. The new integration version works locally, but we're working on getting the test suite running on OTP 21.

 We'll release 1.6.6-beta.1 as soon as we have a working version to allow you to try it out.

**Comment by jeffkreeftmeijer** [comment]
We’ve released version 1.6.6-beta.1 of the AppSignal integration to fix this issue. We’re still doing internal testing to make sure everything works before we release an official 1.6.6.

To test it, update the dependency line in your project's `mix.exs` to accept 1.6.6 beta versions;

```
{:appsignal, "~> 1.6.6-beta"}
```
... and run mix deps.update appsignal.

Please let us know if you run into any issues testing the beta.

**Comment by jeffkreeftmeijer** [comment]
We've released version 1.6.6 of our Elixir integration to fix the issues with OTP 21. To update, please ensure you `mix.exs` file targets the non-beta version of the package;

```
{:appsignal, "~> 1.0"}
```

And run `mix deps.update appsignal`.
